testing/src/main/java/org/oppia/android/testing/media/TestMediaPlayerRobolectricImpl.kt

```diff
@@ -0,0 +1,169 @@
+package org.oppia.android.testing.media
+
+import android.content.Context
+import android.media.MediaPlayer
+import android.net.Uri
+import java.io.IOException
+import javax.inject.Inject
+
+// TODO(#59): Replace the reflection code below with direct calls to Robolectric once this test can be made to run
+//  only on Robolectric (or properly on Espresso without relying on Robolectric shadows, e.g. by using compile-time
+//  replaceable fakes).
+
+/** NOTE TO DEVELOPERS: DO NOT REPLICATE THE REFLECTION CODE BELOW ANYWHERE.
+ * THIS IS A STOP-GAP MEASURE UNTIL WE CAN USE BAZEL TO PROPERLY BUILD THIS TEST SPECIFICALLY
+ * FOR ROBOLECTRIC AND NOT FOR ESPRESSO.
+ *
+ *
+ * Robolectric-specific implementation of [TestMediaPlayer].
+ *
+ * This implementation uses Reflection.
+ */
+class TestMediaPlayerRobolectricImpl @Inject constructor() : TestMediaPlayer {
+
+  override fun addMediaInfo(
+    context: Context,
+    explorationId1: String,
+    explorationId2: String,
+    audioFileName1: String,
+    audioFileName2: String,
+    mediaPlayer: MediaPlayer
+  ) {
+    val mediaTestUrl1 = createAudioUrl(explorationId1, audioFileName1)
+    val mediaTestUrl2 = createAudioUrl(explorationId2, audioFileName2)
+
+    val dataSource = toDataSource(context, Uri.parse(mediaTestUrl1))!!
+    val dataSource2 = toDataSource(context, Uri.parse(mediaTestUrl2))!!
+    val mediaInfo = createMediaInfo(
+      duration = 1000,
+      preparationDelay = 0
```

Comment:
should we specify `duration ` and `preparationDelay ` here only or should we pass as an argument from test case?

Reply:
I think it may be cleaner to define these as arguments with defaults so that the test can choose whether to provide them. It also exposes what the default values are so that it's not hidden information.

Reply:
Done, 
as per the `TestMediaPlayer` interface, need to put default values in the interface.